Make Lara say 'no'  if she has no items when interacting with receptacles

rr- opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

Resolves #3337.

Regression TBD

rr- committed:
tr1/objects: say 'no' if Lara has no key items

Resolves #3337.

lahm86 reviewed (approved):
LGTM, although Lara says No continuously if you hold action. TR2 has some sort of cool-off/position reset IIRC?

rr- commented:
@walkawayy  brought up the issue of interacting with used keyholes.

Scenarios I can think of: Lara next to keyhole, trying to interact with it:

1. Not yet interacted. No items in her ring. Should say 'no', likely
2. Not yet interacted. A matching item in her ring. Should use it
3. Not yet interacted. A non-matching item in her ring. Should use it
4. Used. No items in her ring. I don't know what she should do
5. Used. A matching item in her ring (key of the same type). I don't know what she should do
6. Used. A non-matching item in her ring. I don't know what she should do
7. Not standing near any receptacle at all. Trying to use an item. She should say 'no' 

I think we'll have to separately test keyholes, puzzle holes, scion pedestals and the Midas hand.

aredfan commented:
Regression since [4.0](https://github.com/LostArtefacts/TRX/releases/tag/tr1-4.0).

aredfan commented:
I've added tables below for each of the scenarios outlined. I omitted the scion pedestal as it didn't require much investigation and appears to be behaving as expected. The only scenarios that may need attention are in the 'Puzzle slot' section where I put question marks next to the result.

<details><summary>Keyhole</summary>

| Scenarios                                            | Result                                                                                                    |
|------------------------------------------------------|-----------------------------------------------------------------------------------------------------------|
| Not yet interacted. No items in her ring.            | TombATI: Lara says ""No"", position reset required ✅<br>TR1X: Lara says ""No"", position reset required ✅ |
| Not yet interacted. A matching item in her ring.     | TombATI: Lara opens the door ✅<br>TR1X: Lara opens the door ✅                                             |
| Not yet interacted. A non-matching item in her ring. | TombATI: Lara says ""No"" ✅<br>TR1X: Lara says ""No"" ✅                                                   |
| Used. No items in her ring.                          | TombATI: Lara says ""No"", position reset required ✅<br>TR1X: Lara says ""No"", position reset required ✅                                         |
| Used. A matching item in her ring.                   | TombATI: Lara says ""No"", position reset required ✅<br>TR1X: Lara says ""No"", position reset required ✅ |
| Used. A non-matching item in her ring.               | TombATI: Lara says ""No"", position reset required ✅<br>TR1X: Lara says ""No"", position reset required ✅                                         |
| Not standing near any receptacle at all (action)     | TombATI: Lara says nothing ✅<br>TR1X: Lara says nothing ✅                                                 |
| Not standing near any receptacle at all (inventory)  | TombATI: Lara says nothing ❌<br>TR1X: Lara says ""No"" ✅                                                  |

</details>

<details><summary>Puzzle slot</summary>

| Scenarios                                            | Result                                                                                                    |
|------------------------------------------------------|-----------------------------------------------------------------------------------------------------------|
| Not yet interacted. No items in her ring.            | TombATI: Lara says ""No"", position reset required ✅<br>TR1X: Lara says ""No"", position reset required ✅ |
| Not yet interacted. A matching item in her ring.     | TombATI: Lara opens the door ✅<br>TR1X: Lara opens the door ✅                                             |
| Not yet interacted. A non-matching item in her ring. | TombATI: Lara says ""No"" ✅<br>TR1X: Lara says ""No"" ✅                                                   |
| Used. No items in her ring.                          | TombATI: Lara says nothing ❓<br>TR1X: Lara says nothing ❓                                                 |
| Used. A matching item in her ring.                   | TombATI: Lara says nothing ❓<br>TR1X: Lara says nothing ❓                                                 |
| Used. A non-matching item in her ring.               | TombATI: Lara says nothing ❓<br>TR1X: Lara says nothing ❓                                                 |
| Not standing near any receptacle at all (action)     | TombATI: Lara says nothing ✅<br>TR1X: Lara says nothing ✅                                                 |
| Not standing near any receptacle at all (inventory)  | TombATI: Lara says nothing ❌<br>TR1X: Lara says ""No"" ✅                                                  |

</details>

<details><summary>Midas hand</summary>

| Scenarios                                            | Result                                                                            |
|------------------------------------------------------|-----------------------------------------------------------------------------------|
| Not yet interacted. No items in her ring.            | TombATI: Lara says nothing ❌<br>TR1X: Lara says ""No"", position reset required ✅ |
| Not yet interacted. A matching item in her ring.     | TombATI: Lara makes gold ✅<br>TR1X: Lara makes gold ✅                             |
| Not yet interacted. A non-matching item in her ring. | TombATI: Lara says nothing ❌<br>TR1X: Lara says ""No"" ✅                                        |
| Used. No items in her ring.                          | N/A                                                                               |
| Used. A matching item in her ring.                   | N/A                                                                               |
| Used. A non-matching item in her ring.               | N/A                                                                               |
| Not standing near any receptacle at all (action)     | TombATI: Lara says nothing ✅<br>TR1X: Lara says nothing ✅                         |
| Not standing near any receptacle at all (inventory)  | TombATI: Lara says nothing ❌<br>TR1X: Lara says ""No"" ✅                          |

</details>

rr- commented:
Thank you @aredfan for very detailed testing. I added a change where Lara says 'no' near complete puzzle holes regardless of items possessed to match keyhole behavior.

I also removed some superfluous logic related to puzzle hole activity tests that aren't really needed considering that complete puzzle holes become different objects. I tested it with and without the items, saving and loading and everything seems to be in order.

I made this into two changelog entries – one about missing items in the key items ring, and the other about staying silent near 'complete' keyholes.

aredfan reviewed (approved):
Thank you @rr- , I retested the complete puzzle hole scenarios.  Everything looks good. 👍
